test_pytest_html.py

```diff
@@ -25,22 +25,48 @@ def run(testdir, path='report.html', *args):
     return result, html
 
 
-def assert_summary(html, tests=1, duration=None, passed=1, skipped=0, failed=0,
+def assert_results_by_outcome(html, test_outcome, test_outcome_number,
+                              label=None):
+    # Asserts if the test number of this outcome in the summary matches
+    regex_summary = '(\d)+ {0}'.format(label or test_outcome)
+    int(re.search(regex_summary, html).group(1)) == test_outcome_number
+
+    # Asserts if the generated checkboxes of this outcome matches
+    regex_checkbox = ('<input checked="true" data-test-result="{0}"'
+                      .format(test_outcome))
+    if test_outcome_number == 0:
+        regex_checkbox += ' disabled="true"'
+    assert str(re.search(regex_checkbox, html).group()) == regex_checkbox
+
+    # Asserts if the generated checkboxes of this outcome matches
+    regex_table = ('tbody class=\"{0} '.format(test_outcome))
+    assert len(re.findall(regex_table, html)) or 0 == test_outcome_number
+
+
+def assert_results(html, tests=1, duration=None, passed=1, skipped=0, failed=0,
                    errors=0, xfailed=0, xpassed=0):
-    m = re.search('(\d)+ tests ran in ([\d,.])+ seconds', html)
-    assert int(m.group(1)) == tests
+    # Asserts total amount of tests
+    total_tests = re.search('(\d)+ tests ran', html)
+    assert int(total_tests.group(1)) == tests
+
+    # Asserts tests running duration
+    tests_duration = re.search('([\d,.])+ seconds', html)
```

Comment:
We could move this into the if block, so we don't do the search unless we're verifying the value.